src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx

```diff
@@ -1,36 +1,47 @@
-import { useContext } from 'react';
+import { useContext, useMemo, useState } from 'react';
 import { IconButton } from '@material-ui/core';
 import { Add } from '@material-ui/icons';
 import FilterListIcon from '@material-ui/icons/FilterList';
 import { useParams } from 'react-router';
 import { Link, useHistory } from 'react-router-dom';
-import AccessContext from '../../../../contexts/AccessContext';
-import { IFeatureToggleListItem } from '../../../../interfaces/featureToggle';
-import { getCreateTogglePath } from '../../../../utils/route-path-helpers';
-import ConditionallyRender from '../../../common/ConditionallyRender';
-import { PROJECTFILTERING } from '../../../common/flags';
-import HeaderTitle from '../../../common/HeaderTitle';
-import PageContent from '../../../common/PageContent';
-import ResponsiveButton from '../../../common/ResponsiveButton/ResponsiveButton';
-import FeatureToggleListNew from '../../../feature/FeatureToggleListNew/FeatureToggleListNew';
+import AccessContext from 'contexts/AccessContext';
+import { SearchField } from 'component/common/SearchField/SearchField';
+import ConditionallyRender from 'component/common/ConditionallyRender';
+import { PROJECTFILTERING } from 'component/common/flags';
+import HeaderTitle from 'component/common/HeaderTitle';
+import PageContent from 'component/common/PageContent';
+import ResponsiveButton from 'component/common/ResponsiveButton/ResponsiveButton';
+import FeatureToggleListNew from 'component/feature/FeatureToggleListNew/FeatureToggleListNew';
+import { IFeatureToggleListItem } from 'interfaces/featureToggle';
+import { getCreateTogglePath } from 'utils/route-path-helpers';
 import { useStyles } from './ProjectFeatureToggles.styles';
-import { CREATE_FEATURE } from '../../../providers/AccessProvider/permissions';
-import useUiConfig from '../../../../hooks/api/getters/useUiConfig/useUiConfig';
+import { CREATE_FEATURE } from 'component/providers/AccessProvider/permissions';
+import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
+import classnames from 'classnames';
 
 interface IProjectFeatureToggles {
     features: IFeatureToggleListItem[];
     loading: boolean;
 }
 
-const ProjectFeatureToggles = ({
+export const ProjectFeatureToggles = ({
     features,
     loading,
 }: IProjectFeatureToggles) => {
     const styles = useStyles();
-    const { id } = useParams();
+    const { id } = useParams<{ id: string }>();
     const history = useHistory();
     const { hasAccess } = useContext(AccessContext);
     const { uiConfig } = useUiConfig();
+    const [filter, setFilter] = useState('');
+    useState<IFeatureToggleListItem[]>(features);
+
+    const filteredFeatures = useMemo(() => {
+        const regExp = new RegExp(filter, 'i');
+        return filter
+            ? features?.filter(feature => regExp.test(feature?.name))
```

Comment:
Looks like `features` will always be defined, so we can drop the `?`. Same for `feature.name`.